enterprise/backend/src/metabase_enterprise/workspaces/api.clj

```diff
@@ -181,6 +181,64 @@
    :limit  (request/limit)
    :offset (request/offset)})
 
+;;;; /tables start
+
+(defn- input-tables
+  [graph]
+  (into []
+        (map #(update-keys %
+                           (fn [k]
+                             (case k
+                               :schema :schema-name
+                               :name :table-name
+                               k))))
+        (t2/select [:model/Table :id :schema :name]
+                   :id [:in (map :id (:inputs graph))])))
+
+(defn- outputs-with-transforms
+  [graph]
+  (let [;; duplicated and original data
+        id->transform-data (t2/select-fn->fn :id identity :model/Transform
+                                             :id [:in (concat (map :id (:transforms graph))
+                                                              (map (comp :id :mapping) (:transforms graph)))])
+        s+t->transform-data (into {}
+                                  (map (fn [[_id td]]
+                                         (let [{:keys [schema name]} (:target td)]
+                                           [[schema name] td])))
+                                  id->transform-data)]
+    (mapv
+     (fn [{:keys [schema name mapping] :as _output}]
+       {:global {:schema-name schema
+                 :table-name name}
+        :workspace {:transform-id (:id (s+t->transform-data [(:schema mapping) (:name mapping)]))
+                    :table-id (:id mapping)}})
```

Comment:
Oh, so you're using the global transform to map from the original output table names.

This is wrong.

The transforms may have been edited to have new targets. In future we may be updating the saved graph to update the outputs, but we're not doing that now. If we start doing that we'll probably just pre-compute this entire response.

What you should be doing is the following:

1. Fetch all the workspace transforms.
2. Get their target table.
3. Use the mapping table to look up their equivalent global table.